gcc debug CI jobs are broken due to resource constraints

`linux-gcc5-debug-coverage` and `node-gcc6-debug` jobs are failing on CircleCI due to their container resource class being reduced to `medium` (from the requested `large`).

These two jobs are currently still enabled but have been temporarily un-required for PRs on GitHub.

[Example failure](https://circleci.com/gh/mapbox/mapbox-gl-native/194600):

```
[114/358] Building CXX object CMakeFiles/mbgl-core.dir/src/mbgl/renderer/renderer_impl.cpp.o
FAILED: CMakeFiles/mbgl-core.dir/src/mbgl/renderer/renderer_impl.cpp.o 
/src/build/linux-x86_64/Debug/launch-cxx /usr/lib/ccache/g++-5   -DMBGL_USE_GLES2=1 -DMBGL_WITH_EGL=1 -DRAPIDJSON_HAS_STDSTRING=1 -D_GLIBCXX_USE_CXX11_ABI=1 -I../../../include -I../../../src -I../../../mason_packages/linux-x86_64/swiftshader/2018-05-31/include -I../../../platform/default -I../../../platform/linux -I../../../mason_packages/linux-x86_64/libpng/1.6.25/include/libpng16 -I../../../mason_packages/linux-x86_64/libjpeg-turbo/1.5.0/include -isystem ../../../vendor/earcut.hpp/include -isystem ../../../vendor/expected/include -isystem ../../../vendor/geojson-vt-cpp/include -isystem ../../../vendor/kdbush.hpp/include -isystem ../../../vendor/shelf-pack-cpp/include -isystem ../../../vendor/supercluster.hpp/include -isystem ../../../vendor/unique_resource -isystem ../../../vendor/wagyu/include -isystem ../../../vendor/boost/include -isystem ../../../vendor/geojson.hpp/include -isystem ../../../vendor/geometry.hpp/include -isystem ../../../vendor/icu/include -isystem ../../../vendor/polylabel/include -isystem ../../../vendor/protozero/include -isystem ../../../vendor/rapidjson/include -isystem ../../../vendor/variant/include -isystem ../../../vendor/vector-tile/include -isystem ../../../vendor/nunicode/include -Werror --coverage -ftemplate-depth=1024 -Wall -Wextra -Wshadow -Wnon-virtual-dtor -Wno-variadic-macros -Wno-unknown-pragmas -fext-numeric-literals -g -fPIC -fvisibility=hidden -fvisibility-inlines-hidden   -DCI_BUILD=1 -std=c++14 -MMD -MT CMakeFiles/mbgl-core.dir/src/mbgl/renderer/renderer_impl.cpp.o -MF CMakeFiles/mbgl-core.dir/src/mbgl/renderer/renderer_impl.cpp.o.d -o CMakeFiles/mbgl-core.dir/src/mbgl/renderer/renderer_impl.cpp.o -c ../../../src/mbgl/renderer/renderer_impl.cpp
g++-5: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-5/README.Bugs> for instructions.
[115/358] Building CXX object CMakeFiles/mbgl-core.dir/src/mbgl/renderer/layers/render_fill_extrusion_layer.cpp.o
[116/358] Building CXX object CMakeFiles/mbgl-core.dir/src/mbgl/renderer/layers/render_fill_layer.cpp.o
[117/358] Building CXX object CMakeFiles/mbgl-core.dir/src/mbgl/renderer/layers/render_line_layer.cpp.o
ninja: build stopped: subcommand failed.
Makefile:355: recipe for target 'glfw-app' failed
make: *** [glfw-app] Error 1
Exited with code 2
```

/cc @kkaefer @tmpsantos @mapbox/maps-android @mapbox/maps-ios @mapbox/gl @mapbox/gl-core

--- a/Makefile
+++ b/Makefile
@@ -70,7 +70,8 @@ MACOS_COMPDB_PATH = $(MACOS_OUTPUT_PATH)/compdb/$(BUILDTYPE)
 MACOS_XCODEBUILD = xcodebuild \
 	-derivedDataPath $(MACOS_OUTPUT_PATH) \
 	-configuration $(BUILDTYPE) \
-	-workspace $(MACOS_WORK_PATH)
+	-workspace $(MACOS_WORK_PATH) \
+	-jobs $(JOBS)
 
 ifneq ($(CI),)
 	MACOS_XCODEBUILD += -xcconfig platform/darwin/ci.xcconfig
@@ -209,7 +210,8 @@ IOS_XCODEBUILD_SIM = xcodebuild \
 	-derivedDataPath $(IOS_OUTPUT_PATH) \
 	-configuration $(BUILDTYPE) -sdk iphonesimulator \
 	-destination 'platform=iOS Simulator,name=iPhone 6,OS=latest' \
-	-workspace $(IOS_WORK_PATH)
+	-workspace $(IOS_WORK_PATH) \
+	-jobs $(JOBS)
 
 ifneq ($(CI),)
 	IOS_XCODEBUILD_SIM += -xcconfig platform/darwin/ci.xcconfig

--- a/circle.yml
+++ b/circle.yml
@@ -461,8 +461,8 @@ jobs:
       - image: mbgl/linux-clang-7:a5a3c52107
     working_directory: /src
     environment:
-      LIBSYSCONFCPUS: 4
-      JOBS: 4
+      LIBSYSCONFCPUS: 2
+      JOBS: 2
       BUILDTYPE: Debug
     steps:
       - install-dependencies: { mason: false, ccache: false }
@@ -496,8 +496,8 @@ jobs:
       - image: mbgl/linux-clang-7:a5a3c52107
     working_directory: /src
     environment:
-      LIBSYSCONFCPUS: 4
-      JOBS: 4
+      LIBSYSCONFCPUS: 2
+      JOBS: 2
       BUILDTYPE: Debug
     steps:
       - install-dependencies: { ccache: false }
@@ -524,11 +524,11 @@ jobs:
         default: android-ndk-r19:8e91a7ebab
     docker:
       - image: mbgl/<< parameters.image >>
-    resource_class: large
+    resource_class: xlarge
     working_directory: /src
     environment:
-      LIBSYSCONFCPUS: 2
-      JOBS: 2
+      LIBSYSCONFCPUS: 8
+      JOBS: 8
       BUILDTYPE: Debug
       IS_LOCAL_DEVELOPMENT: false
       MBGL_ANDROID_STL: << parameters.stl >>
@@ -592,11 +592,11 @@ jobs:
   android-release:
     docker:
       - image: mbgl/android-ndk-r19:8e91a7ebab
-    resource_class: large
+    resource_class: xlarge
     working_directory: /src
     environment:
-      LIBSYSCONFCPUS: 2
-      JOBS: 2
+      LIBSYSCONFCPUS: 8
+      JOBS: 8
       BUILDTYPE: Release
       IS_LOCAL_DEVELOPMENT: false
     steps:
@@ -699,6 +699,7 @@ jobs:
   node-clang39-release:
     docker:
       - image: mbgl/linux-clang-3.9:2077f965ed
+    resource_class: large
     working_directory: /src
     environment:
       LIBSYSCONFCPUS: 4
@@ -721,8 +722,8 @@ jobs:
     resource_class: large
     working_directory: /src
     environment:
-      LIBSYSCONFCPUS: 2
-      JOBS: 2
+      LIBSYSCONFCPUS: 4
+      JOBS: 4
       BUILDTYPE: Debug
       WITH_EGL: 1
     steps:
@@ -756,6 +757,7 @@ jobs:
   linux-clang-38-libcxx-debug:
     docker:
       - image: mbgl/linux-clang-3.8-libcxx:d6800bdbb4
+    resource_class: large
     working_directory: /src
     environment:
       LIBSYSCONFCPUS: 4
@@ -772,6 +774,7 @@ jobs:
   linux-clang-7-sanitize-address-undefined:
     docker:
       - image: mbgl/linux-clang-7:a5a3c52107
+    resource_class: large
     working_directory: /src
     environment:
       LIBSYSCONFCPUS: 4
@@ -794,6 +797,7 @@ jobs:
   linux-clang-7-sanitize-thread:
     docker:
       - image: mbgl/linux-clang-7:a5a3c52107
+    resource_class: large
     working_directory: /src
     environment:
       LIBSYSCONFCPUS: 4
@@ -818,8 +822,8 @@ jobs:
     resource_class: large
     working_directory: /src
     environment:
-      LIBSYSCONFCPUS: 2
-      JOBS: 2
+      LIBSYSCONFCPUS: 4
+      JOBS: 4
       BUILDTYPE: Debug
       WITH_EGL: 1
       WITH_CXX11ABI: 0
@@ -839,8 +843,8 @@ jobs:
     resource_class: large
     working_directory: /src
     environment:
-      LIBSYSCONFCPUS: 2
-      JOBS: 2
+      LIBSYSCONFCPUS: 4
+      JOBS: 4
       BUILDTYPE: Debug
       WITH_EGL: 1
       WITH_COVERAGE: 1
@@ -1045,8 +1049,8 @@ jobs:
     resource_class: large
     working_directory: /src
     environment:
-      LIBSYSCONFCPUS: 2
-      JOBS: 2
+      LIBSYSCONFCPUS: 4
+      JOBS: 4
       BUILDTYPE: Release
       WITH_QT_I18N: 1
     steps:

